### PR TITLE
Issues/34: More compact array/map representation

### PR DESCRIPTION
**Pull request by komuw**
# What:
- Where the elements in a slice/array are primitive types; dump in a compact manner.
- Where the elements(not keys) in a map are primitive types; dump in a compact manner

# Why:
- Fixes: https://github.com/sanity-io/litter/issues/34

### CODE REVIEW

File: dump.go [hunk 1]
```diff
@@ -106,6 +106,17 @@ func (s *dumpState) dumpType(v reflect.Value) {
 }
 
 func (s *dumpState) dumpSlice(v reflect.Value) {
+	elmType := v.Type().Elem()
+	switch elmType.Kind() {
+	case reflect.Bool, reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64, reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64, reflect.Uintptr, reflect.Float32, reflect.Float64, reflect.String:
+		// Where the elements in a slice/array are primitive types; dump in a compact manner
+		original := s.config.Compact
+		s.config.Compact = true
```
Comment (atombender) [inline dump.go:114]: We should never mutate the configuration, which might be shared across multiple concurrent goroutines, and might be a constant at the caller's end. Let's add the flag to `dumpState` directly.

**Review by atombender — COMMENTED** [review]
My concern here is wrapping. If you have a slice or map of a certain number of elements, they are going to all be on the same line.

I looked at implementing a compact mode for slices/maps at one point, but ran into the problem that you need to handle wrapping and indentation everywhere, and I didn't bother finishing it. Ideally, indentation and wrapping should be handled at the writer level.

**Comment by komuw** [comment]
I have given this a little more thought. I think the work needed here is a little more than I had imagined. 
I don't think I'll be able to get to it anytime soon; thus I'm electing to close this PR. 

`litter` is a tool that has saved me a lot of minutes and I would like to take this moment to thank all who work on it. Thank you.

**Comment by atombender** [comment]
Thank you! I might revisit this at some point. It's annoying, I agree.

**Pull request closed without merging**